e2e/test/scenarios/question/reproductions/19893-joining-same-table-and-field-multiple-times.cy.spec.js

```diff
@@ -0,0 +1,118 @@
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+import { restore } from "e2e/support/helpers";
+
+const { PRODUCTS, PRODUCTS_ID } = SAMPLE_DATABASE;
+
+const QUESTION_1 = {
+  name: "Q1",
+  query: {
+    "source-table": PRODUCTS_ID,
+    aggregation: [["count"]],
+    breakout: [["field", PRODUCTS.CATEGORY, { "base-type": "type/Text" }]],
+  },
+};
+
+const QUESTION_2 = {
+  name: "Q2",
+  query: {
+    "source-table": PRODUCTS_ID,
+    aggregation: [
+      ["sum", ["field", PRODUCTS.PRICE, { "base-type": "type/Float" }]],
+    ],
+    breakout: [["field", PRODUCTS.CATEGORY, { "base-type": "type/Text" }]],
+  },
+};
+
+describe("issue 19893", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should display correct join source table when joining visited questions (metabase#19893)", () => {
+    cy.createQuestion(QUESTION_1, {
+      wrapId: true,
+      idAlias: "questionId1",
+      visitQuestion: true,
+    });
+    cy.createQuestion(QUESTION_2, {
+      wrapId: true,
+      idAlias: "questionId2",
+      visitQuestion: true,
+    });
+
+    cy.then(function () {
+      const { questionId1, questionId2 } = this;
+
+      createQ1PlusQ2Question(questionId1, questionId2).then(
+        ({ body: question }) => {
+          cy.visit(`/question/${question.id}/notebook`);
+        },
+      );
+    });
+
+    assertQ1PlusQ2Joins();
+  });
+
+  it.skip("should display correct join source table when joining non-visited questions (metabase#19893)", () => {
```

Comment:
This test is still failing, that's why it's skipped.
I didn't add more tests because some assertions are not clear to me (I'm not sure what is the expected result).

Reply:
@kamilmielnik 

For the future - if you're adding a skipped repro add `.skip` to the top level `describe` block rather than to the individual tests.

Also, please add `support/Repro` label to it, and label the underlying issue with `.Reproduced`